<launch>
    <!-- name of polygon without ending -->
    <arg name="poly_name" value="table1" />

    <!-- select group_name: left_arm or right_arm or combined-->
    <arg name="group_name" value="right_arm" />
    
    <!-- resolution in m (0.1, 0.05, 0.025) -->
    <arg name="r" value="0.1" />
    <!-- theta_resolution (unsigned int) -->
    <arg name="t" value="16" />

	<arg name="x-robot" value="0.0" />
	<arg name="y-robot" value="-0.45" />
	<!-- theta index -->
	<arg name="theta-robot" value="8" />
	<arg name="z-offset" value="-0.11" />
    

	<!-- DO NOT CHANGE PARAMETERS BELOW -->
	   
	<!-- path to inverse capability map -->
    <arg name="path_name" value="$(find inverse_capability_map_generator)/maps/inv_cap_$(arg poly_name)_pr2_$(arg group_name)_$(arg r)m_$(arg t)t.icpm" />
    <!-- path to polygon -->
    <arg name="path_poly" value="$(find primitive_to_polygon)/polygons/$(arg poly_name).poly" />
	    
	<include file="$(find tidyup_pr2_moveit_config)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>

    <node pkg="inverse_capability_map" type="inverse_capability_polygon_robot_vis" name="inverse_capability_polygon_robot_vis"
    	args=" -p $(arg path_name)
    	-i $(arg path_poly)
    	-x $(arg x-robot) -y $(arg y-robot) -t $(arg theta-robot) -z $(arg z-offset)"
		output="screen">
		<param name="poly_name" type="string" value="$(arg poly_name)" />
    </node>
</launch>
